tests/all/cli_tests/threads.wat

```diff
@@ -0,0 +1,62 @@
+(module
+  ;; As we have discussed, it makes sense to make the shared memory an import
+  ;; so that all
+  (import "" "memory" (memory $shmem 1 1 shared))
+  (import "wasi_snapshot_preview1" "fd_write"
+    (func $__wasi_fd_write (param i32 i32 i32 i32) (result i32)))
+  (import "wasi_snapshot_preview1" "proc_exit"
+    (func $__wasi_proc_exit (param i32)))
+  (import "wasi" "thread_spawn"
+    (func $__wasi_thread_spawn (param i32) (result i32)))
+
+  (func (export "_start")
+    (local $i i32)
+
+    ;; Print "Called _start".
+    (call $print (i32.const 32) (i32.const 14))
+
+    ;; Print "Running wasi_thread_start" in several threads.
+    (drop (call $__wasi_thread_spawn (i32.const 0)))
+    (drop (call $__wasi_thread_spawn (i32.const 0)))
+    (drop (call $__wasi_thread_spawn (i32.const 0)))
+
+    ;; Wait for all the threads to notify us that they are done.
+    (loop $again
+      ;; Retrieve the i32 at address 128, compare it to -1 (it should always
+      ;; fail) and load it atomically to check if all three threads are
+      ;; complete. This wait is for 1ms or until notified, whichever is first.
+      (drop (memory.atomic.wait32 (i32.const 128) (i32.const -1) (i64.const 1000000)))
```

Comment:
if i read this correctly, this `wait` would always return 1 immediately without sleeping.

Reply:
Ah, yeah, this is a mistake due to a refactoring: originally I intended the `i` local (unused now) to keep track of the expected values but that was lost as I changed things. I'll add a note to fix this...

Reply:
Fixed in #5858